datafusion/sql/src/statement.rs

```diff
@@ -710,6 +710,19 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
                     *statement,
                     &mut planner_context,
                 )?;
+
+                if data_types.is_empty() {
```

Comment:
Over all looks good to me, minor suggestion:

```suggestion
                if data_types.is_empty() {
                     let map_types = plan.get_parameter_types()?;
                     let param_types: Vec<_> = (1..=map_types.len())
                        .filter_map(|i| {
                             let key = format!("${i}");
                             map_types
                                .get(&key)
                                .and_then(|opt| opt.clone())  // gives Option<DataType>
                        })
                        .collect();
                     data_types.extend(param_types.iter().cloned());
                     planner_context.with_prepare_param_data_types(param_types);
                  }
```

This way we can avoid using `.unwrap()` and using `if let` to extract `value` from `data_type`.

Reply:
This looks more rust idiomatic and avoids the unwrap so I will try this, thanks.